ding/reward_model/trex_reward_model.py

```diff
@@ -0,0 +1,618 @@
+                    gt_rewards = []
+                    data_for_save = {}
+                    self.data_for_save[int(checkpoint) // int(checkpoint_step)] = []
+                    r = 0
+                    env.seed(self.cfg.seed + (int(checkpoint) - int(checkpoint_min)) // int(checkpoint_step))
+                    ob = env.reset()
+                    steps = 0
+                    acc_reward = 0
+                    while True:
+                        obs_tensor = torch.tensor(ob).unsqueeze(0)
+                        logit = model.forward(obs_tensor.float(), mode='compute_actor_critic')['logit']
+                        dist = Categorical(logits=logit)
+                        #action = dist.argmax(dim=-1)
+                        action = dist.sample()
+                        action = action.numpy()
+                        if action.shape == (1, ):
+                            action = action.squeeze()  # 0-dim array
+                        ob, r, done, _ = env.step(action)
+
+                        # save data
+                        data_for_save['obs'] = obs_tensor.squeeze().float()
+                        data_for_save['next_obs'] = torch.Tensor(ob)
+                        data_for_save['action'] = torch.Tensor(np.expand_dims(action, axis=0))
+                        data_for_save['reward'] = torch.Tensor([r])
+                        data_for_save['done'] = done
+                        self.data_for_save[int(checkpoint) // int(checkpoint_step)].append(deepcopy(data_for_save))
+
+                        if isinstance(self.cfg.policy.model.get('obs_shape'), int) or len(
+                                self.cfg.policy.model.get('obs_shape')) == 1:
+                            ob_processed = ob
+                        elif len(self.cfg.policy.model.get('obs_shape')) == 3:
+                            ob_processed = torch.tensor(ob).permute(1, 2, 0).numpy()
+                        traj.append(ob_processed)
+                        gt_rewards.append(r)
+                        steps += 1
+                        acc_reward += r
+                        if done:
+                            print("checkpoint: {}, steps: {}, return: {}".format(checkpoint, steps, acc_reward))
+                            break
+                    print("traj length", len(traj))
+                    print("demo length", len(self.pre_expert_data))
+                    self.pre_expert_data.append(traj)
+                    self.learning_returns.append(acc_reward)
+                    self.learning_rewards.append(gt_rewards)
+
+        return self.pre_expert_data, self.learning_returns, self.learning_rewards
+
+    def generate_novice_demos_continuous_action_space(self):
+        env = wrap_mujoco(
+            self.cfg.env.env_id,
+            norm_obs=self.cfg.env.get('norm_obs', None),
+            norm_reward=self.cfg.env.get('norm_reward', None),
+            only_info=False
+        )
+        checkpoint_min = 1000
```

Comment:
don't hardcode them, at least set them as the argument of this method

Reply:
Thx, it is done